vault/resource_consul_secret_backend_role.go

```diff
@@ -93,78 +144,67 @@ func consulSecretBackendRoleRead(d *schema.ResourceData, meta interface{}) error
 
 	data := secret.Data
 	d.Set("name", name)
-	d.Set("path", path)
+	d.Set("backend", backend)
 	d.Set("policies", data["policies"])
+	d.Set("max_ttl", data["max_ttl"])
+	d.Set("ttl", data["ttl"])
+	d.Set("token_type", data["token_type"])
+	d.Set("local", data["local"])
 
 	return nil
 }
 
-func consulSecretBackendRoleUpdate(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*api.Client)
-
-	s := strings.Split(d.Id(), ",")
-	path := s[0]
-	name := s[1]
-
-	reqPath := consulSecretBackendRolePath(path, name)
-
-	d.Partial(true)
-
-	if d.HasChange("policies") {
-		log.Printf("[DEBUG] Updating role configuration at %q", reqPath)
-		policies := d.Get("policies").([]interface{})
-
-		payload := map[string]interface{}{
-			"policies": policies,
-		}
-		if _, err := client.Logical().Write(reqPath, payload); err != nil {
-			return fmt.Errorf("error writing role configuration for %q: %s", reqPath, err)
-		}
-		log.Printf("[DEBUG] Updated role configuration at %q", reqPath)
-		d.SetPartial("policies")
-	}
-
-	d.Partial(false)
-	return consulSecretBackendRoleRead(d, meta)
-}
-
 func consulSecretBackendRoleDelete(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*api.Client)
 
-	s := strings.Split(d.Id(), ",")
-	path := s[0]
-	name := s[1]
-
-	reqPath := consulSecretBackendRolePath(path, name)
+	path := d.Id()
 
-	log.Printf("[DEBUG] Deleting Consul backend role at %q", reqPath)
+	log.Printf("[DEBUG] Deleting Consul backend role at %q", path)
 
-	if _, err := client.Logical().Delete(reqPath); err != nil {
-		return fmt.Errorf("error deleting Consul backend role at %q: %s", reqPath, err)
+	if _, err := client.Logical().Delete(path); err != nil {
+		return fmt.Errorf("error deleting Consul backend role at %q: %s", path, err)
 	}
-	log.Printf("[DEBUG] Deleted Consul backend role at %q", reqPath)
+	log.Printf("[DEBUG] Deleted Consul backend role at %q", path)
 	return nil
 }
 
 func consulSecretBackendRoleExists(d *schema.ResourceData, meta interface{}) (bool, error) {
 	client := meta.(*api.Client)
 
-	s := strings.Split(d.Id(), ",")
-	path := s[0]
-	name := s[1]
-
-	reqPath := consulSecretBackendRolePath(path, name)
+	path := d.Id()
 
-	log.Printf("[DEBUG] Checking Consul secrets backend role at %q", reqPath)
+	log.Printf("[DEBUG] Checking Consul secrets backend role at %q", path)
 
-	secret, err := client.Logical().Read(reqPath)
+	secret, err := client.Logical().Read(path)
 	if err != nil {
-		return false, fmt.Errorf("error reading role configuration for %q: %s", reqPath, err)
+		return false, fmt.Errorf("error reading role configuration for %q: %s", path, err)
 	}
 
 	return secret != nil, nil
 }
 
-func consulSecretBackendRolePath(path, name string) string {
-	return strings.Trim(path, "/") + "/roles/" + name
+func consulSecretBackendRolePath(backend, name string) string {
+	return strings.Trim(backend, "/") + "/roles/" + name
+}
+
+func consulSecretBackendRoleNameFromPath(path string) (string, error) {
```

Comment:
Would probably be good to get a unit test on this one.

Reply:
Done